src/mmrelay/meshtastic/packet_routing.py

```
@@ -0,0 +1,288 @@
+class PacketAction:
+    RELAY = "relay"
+    PLUGIN_ONLY = "plugin_only"
+    DROP = "drop"
+
+
+CHAT_ELIGIBLE_PORTNUMS = frozenset({TEXT_MESSAGE_APP})
+logger = get_logger("Meshtastic")
+_warned_packet_routing_issues: set[str] = set()
+
+
+def _warn_once(key: str, message: str, *args: Any) -> None:
```

Ruff (0.15.9)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `*args`

(ANN401)